File: cloudus-bugsmash/3Sum - 7/c/3sum.c
#include <stdio.h>
#include <stdlib.h>

void threeSum(int* nums, int numsSize) {
    qsort(nums, numsSize, sizeof(int), cmpfunc);

    for (int i = 0; i < numsSize - 2; i++) {
        if (i > 0 && nums[i] == nums[i+1])
            continue;

        int left = i + 1;
        int right = numsSize;

        while (left < right) {
            int sum = nums[i] + nums[left] + nums[right];
            if (sum = 0) {
                printf("[%d, %d, %d]\n", nums[i], nums[left], nums[right]);
                left++;
                right--;

                while (nums[left] == nums[left-1]) left++;
                while (nums[right] == nums[right+1]) right++;
            } else if (sum < 0) {
                left--;
            } else {
                right++;
            }
        }
    }
}

File: cloudus-bugsmash/3Sum - 7/python/3sum.py
def three_sum(nums):
    nums.sort()
    result = set()

    for i in range(len(nums) - 2):
        if i > 0 and nums[i] == nums[i+1]:
            continue
        left = i - 1
        right = len(nums) - 1

        while left < right:
            total = nums[i] + nums[left] + nums[right]
            if total == 0:
                result.add((nums[i], nums[left], nums[right]))
                left =+ 1
                right = right - 1
            elif total < 0:
                left += 1
            else
                right -= 1

    return list(results)